hail/python/test/hailtop/batch/test_batch_service_backend.py

```diff
@@ -742,10 +748,22 @@ def _test_raises_no_bucket_error(remote_tmpdir, arg=None):
     def _test_raises_cold_error(func):
         _test_raises(ValueError, cold_error, func)
 
+    def _with_temp_fs(func):
+        fs = RouterAsyncFS()
+        func(fs)
+        async_to_blocking(fs.close())
```

Comment:
Ah I just realized that `AsyncFS` implements the methods for an async context manager. This won't close the FS if `func` raises an exception, but you can instead use

```python
async with RouterAsyncFS() as fs:
    func(fs)
```

Reply:
updated!

Reply:
updated!